extractor/src/test/suite/scala/com/digitalasset/extractor/config/ConfigParserSpec.scala

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.extractor.config
+
+import com.digitalasset.extractor.config.Generators._
+import com.digitalasset.extractor.targets.Target
+import org.scalacheck.Gen
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+import org.scalatest.{FlatSpec, Inside, Matchers}
+
+class ConfigParserSpec
+    extends FlatSpec
+    with Matchers
+    with Inside
+    with GeneratorDrivenPropertyChecks {
+  behavior of ConfigParser.getClass.getSimpleName
+
+  val requiredArgs = Vector("--party", "Bob")
+
+  it should "parse template configuration" in forAll(Gen.nonEmptyListOf(genTemplateConfig)) {
+    templateConfigs: List[TemplateConfig] =>
+      val args = requiredArgs ++ Vector("--templates", templateConfigUserInput(templateConfigs))
+      inside(ConfigParser.parse(args)) {
+        case Some((config, _)) =>
+          config.templateConfigs should ===(templateConfigs.toSet)
+      }
+  }
+
+  it should "fail parsing when duplicate template configurations" in forAll(
+    Gen.nonEmptyListOf(genTemplateConfig)) { templateConfigs: List[TemplateConfig] =>
```

Comment:
scalaz ScalaCheckBinding includes an `Arbitrary` for `OneAnd[List, ?]` if you want to skip the runtime check below. (There's no benefit here to using the explicit `Gen` passing to `forAll` instead of implicit `Arbitrary` resolution.)